storage/Filesystems/BtrfsImpl.cc

```diff
@@ -241,11 +293,96 @@ namespace storage
     }
 
 
+    bool
+    Btrfs::Impl::equal(const Device::Impl& rhs_base) const
+    {
+	const Impl& rhs = dynamic_cast<const Impl&>(rhs_base);
+
+	if (!BlkFilesystem::Impl::equal(rhs))
+	    return false;
+
+	return metadata_raid_level == rhs.metadata_raid_level && data_raid_level == rhs.data_raid_level;
+    }
+
+
+    void
+    Btrfs::Impl::log_diff(std::ostream& log, const Device::Impl& rhs_base) const
+    {
+	const Impl& rhs = dynamic_cast<const Impl&>(rhs_base);
+
+	BlkFilesystem::Impl::log_diff(log, rhs);
+
+	storage::log_diff_enum(log, "metadata-raid-level", metadata_raid_level, rhs.metadata_raid_level);
+	storage::log_diff_enum(log, "data-raid-level", data_raid_level, rhs.data_raid_level);
+    }
+
+
+    void
+    Btrfs::Impl::print(std::ostream& out) const
+    {
+	BlkFilesystem::Impl::print(out);
+
+	out << " metadata-raid-level:" << toString(metadata_raid_level) << " data-raid-level:"
+	    << toString(data_raid_level);
+    }
+
+
+    void
+    Btrfs::Impl::probe_btrfses(Prober& prober)
+    {
+	if (!support_btrfs_multiple_devices())
+	    return;
+
+	SystemInfo& system_info = prober.get_system_info();
+	Devicegraph* system = prober.get_system();
+
+	const CmdBtrfsFilesystemShow& cmd_btrfs_filesystem_show = system_info.getCmdBtrfsFilesystemShow();
+
+	for (const CmdBtrfsFilesystemShow::value_type& tmp : cmd_btrfs_filesystem_show)
```

Comment:
`tmp` is not a good variable name for this; this is the loop variable for this entire longish loop. There must be a better name for it. `btrfs` comes to mind. Or maybe `detected_btrfs`. But this variable is really the pivotal point of this whole block, so it deserves some more attention, i.e. a speaking name.